stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -326,19 +326,41 @@ julia> kron(A, B)
  3+0im  0-3im  4+0im  0-4im
 ```
 """
-function kron(a::AbstractMatrix{T}, b::AbstractMatrix{S}) where {T,S}
-    require_one_based_indexing(a, b)
-    R = Matrix{promote_op(*,T,S)}(undef, size(a,1)*size(b,1), size(a,2)*size(b,2))
+function kron(A::AbstractMatrix{T}, B::AbstractMatrix{S}) where {T,S}
+    R = Matrix{promote_op(*,T,S)}(undef, size(A,1)*size(B,1), size(A,2)*size(B,2))
+    return @inbounds kron!(R, A, B)
+end
+
+"""
+    kron!(C, A, B)
+
+Like [`kron`](@ref), but stores the results in `C`.
+
+!!! tip
+
+    Bounds checking can be disabled by `@inbounds`, but you need to take care of the shape
+    of `C`, `A`, `B` youself.
+"""
+@inline Base.@propagate_inbounds function kron!(C::AbstractMatrix, A::AbstractMatrix, B::AbstractMatrix)
+    require_one_based_indexing(A, B)
+
+    mA, nA = size(A); mB, nB = size(B); mC, nC = size(C)
+    @boundscheck mC == mA * mB && nC == nA * nB || throw(DimensionMismatch("A has size($mA, $nA), B has size($mB, $nB), C has size ($mC, $nC)"))
```

Comment:
Generally speaking we try to keep line lengths at 92 characters. This line is long enough that it necessitates horizontal scrolling in the GitHub UI. I would just turn the `condition || error` form into a long-form `if`. (Note that this same comment applied in a number of places; I won't comment on each individually.)